packages/run-protocol/src/vaultFactory/vaultDirector.js

```diff
@@ -74,260 +81,296 @@ const initState = (zcf, electorateParamManager, debtMint) => {
-
-  // TODO put on machineFacet for use again in publicFacet
-  const getCollaterals = async () => {
-    // should be collateralTypes.map((vm, brand) => ({
-    return harden(
-      Promise.all(
-        [...collateralTypes.entries()].map(async ([brand, vm]) => {
-          const priceQuote = await vm.getCollateralQuote();
-          return {
-            brand,
-            interestRate: vm.getGovernedParams().getInterestRate(),
-            liquidationMargin: vm.getGovernedParams().getLiquidationMargin(),
-            stabilityFee: vm.getGovernedParams().getLoanFee(),
-            marketPrice: makeRatioFromAmounts(
-              getAmountOut(priceQuote),
-              getAmountIn(priceQuote),
-            ),
-          };
-        }),
-      ),
+  /** @param {ZCFSeat} seat */
+  const makeVaultHook = async seat => {
+    assertProposalShape(seat, {
+      give: { Collateral: null },
+      want: { RUN: null },
+    });
+    const {
+      give: { Collateral: collateralAmount },
+    } = seat.getProposal();
+    const { brand: brandIn } = collateralAmount;
+    assert(
+      collateralTypes.has(brandIn),
+      X`Not a supported collateral type ${brandIn}`,
     );
+    /** @type {VaultManager} */
+    const mgr = collateralTypes.get(brandIn);
+    return mgr.makeVaultKit(seat);
   };
+  return zcf.makeInvitation(makeVaultHook, 'MakeVault');
+};
 
-  /** @type {VaultFactory} */
-  const machineBehavior = {
-    // TODO move this under governance #3924
-    /** @type {AddVaultType} */
-    addVaultType: async (
-      collateralIssuer,
-      collateralKeyword,
-      initialParamValues,
-    ) => {
-      await zcf.saveIssuer(collateralIssuer, collateralKeyword);
-      const collateralBrand = zcf.getBrandForIssuer(collateralIssuer);
-      // We create only one vault per collateralType.
-      assert(
-        !collateralTypes.has(collateralBrand),
-        `Collateral brand ${collateralBrand} has already been added`,
-      );
+// TODO put on machineFacet for use again in publicFacet
+/**
+ * Make a loan in the vaultManager based on the collateral type.
+ *
+ * @deprecated
+ * @param {MethodContext} context
+ */
+const getCollaterals = async ({ state }) => {
+  const { collateralTypes } = state;
+  // should be collateralTypes.map((vm, brand) => ({
+  return harden(
+    Promise.all(
+      [...collateralTypes.entries()].map(async ([brand, vm]) => {
+        const priceQuote = await vm.getCollateralQuote();
+        return {
+          brand,
+          interestRate: vm.getGovernedParams().getInterestRate(),
+          liquidationMargin: vm.getGovernedParams().getLiquidationMargin(),
+          stabilityFee: vm.getGovernedParams().getLoanFee(),
+          marketPrice: makeRatioFromAmounts(
+            getAmountOut(priceQuote),
+            getAmountIn(priceQuote),
+          ),
+        };
+      }),
+    ),
+  );
+};
+
+/** @type {import('@agoric/vat-data/src/types').FunctionsPlusContext<VaultFactory>} */
+const machineBehavior = {
```

Comment:
the functions within this behavior could be at module-scope. That's how @FUDCo had in the initial defineKind-ization of vault.js. I propose a style of defining behavior objects with the methods inlined so it's easier to navigate.

Reply:
I had originally had the behavior functions in `vault.js` at module scope simply because it was the most similar to what had been there before.  @turadg asked me to rewrite them to be inline and I think I like the inline style better in that case.  I don't know how it translates to this file however.